safehttp/mux.go

```diff
@@ -97,14 +97,33 @@ func NewServeMux(d Dispatcher, domains ...string) *ServeMux {
 
 // Handle registers a handler for the given pattern and method. If another
 // handler is already registered for the same pattern and method, Handle panics.
-func (m *ServeMux) Handle(pattern string, method string, h Handler) {
+//
+// Configs can be optionally passed which will modify the behavior of the
+// interceptors on the handler registered.
+func (m *ServeMux) Handle(pattern string, method string, h Handler, configs ...Config) {
+	interceps := map[string]Interceptor{}
+	for k, i := range m.interceps {
+		interceps[k] = i
+	}
+	for _, c := range configs {
+		for k, i := range interceps {
+			if i, ok := c.Apply(i); ok {
+				interceps[k] = i
+			}
+		}
+	}
```

Comment:
Do we need to copy the map over if no configs have been passed?
E.g. if we never get `true` we could just discard the new map.